File: src/stringlen.f
	integer function stringlen(string)
c
c	Returns actual length of string, assuming the end is padded
c	with spaces.

	character*(*) string

	length=len(string)


1	if(string(length:length).eq.' ') then
		length=length-1
		if(length.gt.0)	goto 1
	endif

	stringlen=length
	return
	end

File: py/stringlen.py
def stringlen(string):
    """
    Returns the actual length of the string, assuming the end is padded with spaces.
    
    Parameters:
    string : str
        The input string, possibly padded with spaces at the end.
    
    Returns:
    int
        The length of the string without trailing spaces.
    """
    length = len(string)
    
    while length > 0 and string[length - 1] == ' ':
        length -= 1
    
    return length
